plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,214 @@
+          time_elements_tuple=time_elements_tuple)
+      except ValueError:
+        parser_mediator.ProduceExtractionWarning('timestamp \"{0!s}\" seems '
+          'invalid - skipping malformed transcript'.format(time_elements_tuple))
+        continue
+      start_time.is_local_time = True
+      event = time_events.DateTimeValuesEvent(start_time,
+        definitions.TIME_DESCRIPTION_START)
+      # we don't use line breaks here since they aren't displayed nicely in TS
+      if full_headers:
+        command = '; '.join(transcript[18:len(transcript)])
+      else:
+        command = '; '.join(transcript[3:len(transcript)])
+      event_data.command = command
+      # finally create event if values have been set
+      if (event is not None and event_data.command is not None and
+            event_data.command != ''):
+        parser_mediator.ProduceEventWithEventData(event, event_data)
+      else:
+        parser_mediator.ProduceExtractionWarning('skipping transcript {0!s} - '
+          'since relevant event values could not be set'.format(transcript))
+
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses an PowerShell transcript file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+
+    data = file_object.read(self._HEADER_READ_SIZE)
+    if not data == b'\xef\xbb\xbf**********************':
+      raise errors.WrongParser('Not a PowerShell transcript file')
+
+    # The current offset of the file-like object needs to point at
+    # the start of the file to parse the data correctly.
+    file_object.seek(0, os.SEEK_SET)
+    file_bytes = file_object.read()
+    # If transcript file was opened in editor, line breaks may have been changed
+    if b'\r\n' in file_bytes:
+      line_bytes = file_bytes.split(b'\r\n')
+    else:
+      line_bytes = file_bytes.split(b'\n')
+    lines = [line.decode('utf-8') for line in line_bytes]
+    sep_indices = []
+    header_data = PowerShellTranscriptEventData()
+    for index, line in enumerate(lines):
+      if index in self.LINE_INFO_DICT:
+        param = self.LINE_INFO_DICT[index]
+        splitted = line.split(': ')
+        if len(splitted) != 2:
+          parser_mediator.ProduceExtractionWarning('line {0!s} seems invalid '
+            'or value empty - skipping header info: {1!s}'.format(line,
+            self.LINE_INFO_DICT[index]))
+          continue
+        # pre-fill data for later objects
+        setattr(header_data, param, splitted[1])
+      if line == self.SEPARATOR:
+        sep_indices.append(index)
+    # the file always starts with a transcript
+    transcript_starts = [0]
+    for i, sep_index in enumerate(sep_indices):
+      # every second separator marks a new transcript
```

Comment:
What if a transcript contains ':' in the text? Would this assumption still hold?

Reply:
A ':' in a transcript should not change anything about this assumption as far as I can see. The ':' character is only used for splitting key and value in the assumed header lines. If you still think that this is an issue, could you give an example?